PVTPackage/source/MultiphaseSystem/PhaseModel/BlackOil/BlackOil_WaterModel.cpp

```diff
@@ -14,18 +14,35 @@
 
 #include "BlackOil_WaterModel.hpp"
 
+#include "Utils/Logger.hpp"
 #include <cmath>
 
 namespace PVTPackage
 {
 
+BlackOil_WaterModel::BlackOil_WaterModel()
+  :
+  m_surfaceMassDensity( 0.0 ),
+  m_surfaceMoleDensity( 0.0 ),
+  m_surfaceMolecularWeight( 0.0 )
+{
+  // constructor called in DeadOilFlash if water is absent
+}
+  
```

Comment:
Does this make sense?
I mean: are we OK to have a `BlackOil_WaterModel` with dummy values set at 0?
That sounds questionable to me.

As per
```
// constructor called in DeadOilFlash if water is absent
```
you cannot know 😉 

The problem is that the way to code is constructed, you have to have a `BlackOil_WaterModel` instance even if there is no water model. Am I correct?
Couldn't we change this design instead?

Reply:
I have not addressed this comment yet. Yes, the problem is comes from the fact that this:
https://github.com/GEOSX/PVTPackage/blob/d24a4886c01fc43d3583effc43b71b6f42034ed3/PVTPackage/source/MultiphaseSystem/PhaseSplitModel/DeadOilFlash.hpp#L48-L50 
forces us to always have the three phase models, even if only two phases are present. I definitely vote for changing the design. Should we have a array of `PhaseModel`s that we would fill at initialization with only the models that we need? How would you like to proceed?